### PR TITLE
[AAP-24296] Add Encrypted field to CredentialTextInput

### PR DESCRIPTION
**Pull request by daphnemaeve**
Adds encrypted field to CredentialTextInput that is shown when field isn't empty but is a secret field.

Screenshot 2024-05-23 at 4 50 47 PM

### CODE REVIEW

**Review by skochay330 — CHANGES_REQUESTED** [review]

**Comment by kialam** [comment]
This is looking really close. I noticed that empty fields have the revert/redo button but in the old UI those buttons are not present unless a field has an encrypted value:
Old UI:
Screenshot 2024-06-07 at 2 27 06 PM


New UI:
Screenshot 2024-06-07 at 2 26 31 PM

File: frontend/awx/access/credentials/CredentialForm.tsx [hunk 1]
```diff
@@ -818,70 +862,133 @@ function CredentialTextInput({
...
+    );
+    setShouldHideField(true);
+  };
 
   useEffect(() => {
-    // mark field as managed by a credential plugin
-    if (accumulatedPluginValues.some((cp) => cp.input_field_name === field.id)) {
-      setValue(field.id, renderFieldValue(field), { shouldDirty: true });
+    if (shouldHideField && isSecretField && isInitialValueEncrypted && !isPromptOnLaunchChecked) {
+      setValue(field.id, t('ENCRYPTED'), { shouldDirty: false });
```
Comment (kialam) [inline frontend/awx/access/credentials/CredentialForm.tsx:880]: This actually updates the stored password value that the API returns (`$encrypted$`) to `ENCRYPTED` and sends it as the new string to the API, which we don't want. We should set it to `initialValues[field.id]` so that it doesn't change the value.

**Review by kialam — CHANGES_REQUESTED** [review]

**Review by marshmalien — APPROVED** [review]
👍 Great work! It's a very complex area of the project. I pulled down and compared the changes locally against the old UI. Looked good!

**Comment by nixocio** [comment]
@marshmalien @kialam I rebased and update credential tests, now there is one credential test failing but it is a known failure due to a cypress command that needs to be updated.

File: cypress/e2e/awx/resources/credentials.cy.ts [hunk 2]
```diff
@@ -613,7 +621,7 @@ describe('Credentials Tabbed View - Job Templates', function () {
     cy.selectDropdownOptionByResourceName('inventory', awxInventory.name);
     cy.selectDropdownOptionByResourceName('project', `${(this.globalProject as Project).name}`);
     cy.selectDropdownOptionByResourceName('playbook', 'hello_world.yml');
-    cy.selectItemFromLookupModal('credential-select', machineCredential.name);
```
Comment (kialam) [inline cypress/e2e/awx/resources/credentials.cy.ts:616]: We should revert this change. `selectItemFromLookupModal ` is the correct command.

**Review by marshmalien — APPROVED** [review]

File: cypress/e2e/awx/resources/credentials.cy.ts (hunk 2 above)
Comment (nixocio) [reply cypress/e2e/awx/resources/credentials.cy.ts]: @kialam reverted.

**Comment by nixocio** [comment]
As a team we decided to merge this one, since the test failures are not related.

**Review by kialam — APPROVED** [review]

**Review by skochay330 — APPROVED** [review]

**Comment by nixocio** [comment]
As a team we decided to merge this one, since the test failures are not related.
